shared/types/emotion.d.ts

```diff
@@ -0,0 +1,10 @@
+import '@emotion/react';
+import { color, shadow } from '@/styles/theme';
+
+export interface CustomTheme {
+    color: typeof color;
+    shadow: typeof shadow;
+}
+declare module '@emotion/react' {
+    export interface Theme extends CustomTheme {}
+}
```

Comment:
`CustomThem`에 대한 타입 추론이 목적이라면 아래처럼 바로 요렇게 해도 될거 같은데
따로 분리해서 상속받는 이유가 있으신가용?
```suggestion
declare module '@emotion/react' {
    export interface CustomTheme {
        color: typeof color;
        shadow: typeof shadow;
    }
}
```

Reply:
https://emotion.sh/docs/typescript#define-a-theme
공식문서에서 사용하는 방식을 따라해보았습니다! 
declare module 안에 `Theme`라는 명칭이 아니라면 저희가 component에서 사용하는 `background: ${({ theme }) => theme.color.background.range};` 코드에서 theme가 인식되지 않아요!

Reply:
아항 그럼 Theme으로 하면 되지 않을까요? 
저희는 third library를 사용하지 않기도 하구요. 내부에 member 타입들이 없어서 타입에러가 뜨네용,,!
`An interface declaring no members is equivalent to its supertype.eslint[@typescript-eslint/no-empty-interface](https://typescript-eslint.io/rules/no-empty-interface)`

Reply:
아하! 에러가 뜨는지 에디터 상에서 확인이 되지 않아서 몰랐네요....! 추후 수정하겠습니다!